Update UI to be a subtree of the Antora default UI (#1906)
LiskHQ/lisk-docs: ui/docs/modules/ROOT/pages/add-static-files.adoc

Although these files and folders end up in the UI bundle, they are not copied into the site's output directory.
In other words, they are not static files.
Rather, they're used to compose the HTML pages that are published or for configuration.

Any files or folders from the UI bundle that Antora does not recognize are classified as static.
Static files and folders are automatically copied from the UI bundle to the UI output directory (e.g., [.path]_++_++_) under the site output directory, preserving the relative path of the file in the UI bundle.
When using the default UI, this list includes the [.path]_css_, [.path]_font_ (generated), [.path]_img_, and [.path]_js_ folders.
Any files or folders that begin with a dot (i.e., hidden) are ignored unless configured otherwise.

Antora's default UI already has several static folders.
Let's set up another folder and add some top-level static files as well.

First, create a folder under [.path]_src_ named [.path]_static_ for the purpose of demonstration:

 $ mkdir src/static

Let's now add a file to this new folder and three static files (one hidden) directly to the [.path]_src_ folder.

> Here's a file tree that shows the location of these new files:

Reviewer: ```suggestion
A file tree that shows the location of these new files can be seen below:
```

....
src/
  static/
    notice.txt
  .DS_Store
  favicon.ico
  favicon.png
....

Let's now consider how the static files flow from the UI project to the published site.

> We'll start with where these files reside within the UI project (non-static files and folders not shown here):

Reviewer: ```suggestion
We'll start with where these files reside within the UI project (non-static files and folders are not shown here):
```

  ...
js/
  site.js
static/
  notice.txt
favicon.ico
favicon.png
....

Notice the absence of the [.path]_.DS_Store_ file, which is hidden.
The [.path]_font_ folder is introduced by the UI build.

When using this UI bundle in an Antora build, the contents of the output directory of your site will look like this:

.Contents of site output directory
....
_/
  favicon.ico
  css/
    site.css
  font/
    roboto-latin-400-italic.woff
    ...
  img/
    back.svg
    ...
  js/
    site.js
  static/
    about.txt
  favicon.ico
  favicon.png
component-a/
  ...
sitemap.xml
...
....

As stated earlier, the static files are placed into the UI output directory (e.g., [.path]_++_++_) in the published site by default.
In this directory, we see the [.path]_css_, [.path]_font_, [.path]_img_ and [.path]_js_ folders, which are static folders provided by the default UI.
We also see the [.path]_static_ folder and the favicon files we added to the UI project in the first step.
We do not see the [.path]_helpers_, [.path]_partials_, and [.path]_layouts_ folders since they are not static files.

If the UI output directory is where you want static files to end up, there's nothing else you have to do.

> If you want all or some of the static files to be moved to the site root, you need to add additional configuration to promote them.

Reviewer: ```suggestion
If you want all or some of the static files to be moved to the site root, you need to add an additional configuration to promote them.
```

== Promote static files

> If you want to promote certain static files to the root of your site, you need to identify them.

Reviewer: ```suggestion
If you want to promote certain static files to the root of your site, it is necessary to identify them.
```

> You identify them using the [.path]_ui.yml_ file.

Reviewer: ```suggestion
They can be identified using the [.path]_ui.yml_ file.
```

Let's promote that one as well.

You can identify all files in a folder using the wildcard `+*+` in the last path segment, such as `+folder/*+`.
You can identify all files in a folder at any depth using the wildcard `+**+` in the last path segment, such as `+folder/**+`.
Matching a folder has no effect (e.g., `folder`).
Wildcards never match hidden files.
Hidden files must always be written using an exact path match.

Let's also promote all files in the [.path]_static_ folder by adding the wildcard match the `static_files` key in the [.path]_ui.yml_ file.

.src/ui.yml
....
static_files:
- favicon*
- static/*
....

Using this configuration, the [.path]_static_ folder will end up at the site root, adjacent to the favicon files, instead of inside the UI output directory.
Notice that the [.path]_static_ folder is copied too, not just its contents.

Now that the static files are where you want them, let's look at how to reference them from the HTML pages.

== Use the static files

Often when you add static files to your site, you need to reference them somewhere.
In the case of a favicon image, it must be referenced in the `<head>` of the HTML page.

> If you are referencing a promoted static file, you'll use the prefix `+{{{siteRootPath}}}+`.

Reviewer: ```suggestion
If you are referencing a promoted static file, use the following prefix `+{{{siteRootPath}}}+`.
```

> Otherwise, you'll use the prefix `+{{{uiRootPath}}}+`.

Reviewer: ```suggestion
Otherwise, it will be necessary to use the following prefix `+{{{uiRootPath}}}+`.
```
